### PR TITLE
Hsm ssl

### PR DESCRIPTION
**Pull request by MaggieNgWu**

### CODE REVIEW

File: cmake/ProjectSDF.cmake [hunk 1]
```diff
@@ -31,8 +31,8 @@ ExternalProject_Add(libsdf
     PREFIX ${CMAKE_SOURCE_DIR}/deps
     DOWNLOAD_NAME sdf.zip
     DOWNLOAD_NO_PROGRESS 1
-    URL https://github.com/WeBankBlockchain/hsm-crypto/archive/refs/heads/v1.0.0.zip   
-    URL_HASH SHA256=b97edd8c099097a03988f674b0091b801b6f1496210aedc066cede2fd8c7c430
+    URL https://github.com/MaggieNgWu/sdf-crypto-1/archive/d2635acc04a1eacd558e734a0603ea0077105f9e.zip
```
Comment (cyjseagull) [inline cmake/ProjectSDF.cmake:34]: Pls not use personal url

File: libdevcrypto/sm4/sm4.cpp [hunk 2]
```diff
@@ -23,10 +23,18 @@
 
 #include "sm4.h"
 #include <cstring>
+#ifdef FISCO_SDF
+#include <stdlib.h>
```
Comment (cyjseagull) [inline libdevcrypto/sm4/sm4.cpp:27]: This does not require macro control。

File: libinitializer/SecureInitializer.cpp [hunk 3]
```diff
@@ -381,6 +389,140 @@ ConfigResult initGmConfig(const boost::property_tree::ptree& pt)
     std::string caPath = dataPath + "/" + pt.get<std::string>(sectionName + ".ca_path", "");
     std::string enKey = dataPath + pt.get<std::string>(sectionName + ".en_key", "gmennode.key");
     std::string enCert = dataPath + pt.get<std::string>(sectionName + ".en_cert", "gmennode.crt");
+#ifdef FISCO_SDF
```
Comment (cyjseagull) [inline libinitializer/SecureInitializer.cpp:392]: it is recommended to abstract into a separate function.

File: libdevcrypto/hsm/HSMSignature.cpp [hunk 4]
```diff
@@ -81,8 +95,12 @@ std::shared_ptr<crypto::Signature> dev::crypto::SDFSM2Sign(
         key, hsm::SM2, (const unsigned char*)hashResult, 32, signature.data(), &signLen);
     if (code != SDR_OK)
     {
-        throw provider.GetErrorMessage(code);
+        CRYPTO_LOG(ERROR) << "[HSMSignature::sign] ERROR of sign"
+                          << LOG_KV("error", provider.GetErrorMessage(code));
+        return nullptr;
     }
+    // cout<<"hash "<< toHex(hashResult)<<endl;
+    cout << "Signature = " << toHex(signature.data(), signature.data() + 64, "") << endl;
```
Comment (ywy2090) [inline libdevcrypto/hsm/HSMSignature.cpp:103]: Do not use std::cout in your code，use LOG instead

**Review by ywy2090 — APPROVED** [review]

File: cmake/ProjectSDF.cmake (hunk 1 above)
Comment (MaggieNgWu) [reply cmake/ProjectSDF.cmake]: Sure

File: libdevcrypto/sm4/sm4.cpp (hunk 2 above)
Comment (MaggieNgWu) [reply libdevcrypto/sm4/sm4.cpp:27]: OK

File: libdevcrypto/hsm/HSMSignature.cpp [hunk 5]
```diff
@@ -43,9 +44,18 @@ std::shared_ptr<crypto::Signature> dev::crypto::SDFSM2Sign(
 {
     CryptoProvider& provider = SDFCryptoProvider::GetInstance();
     Key key = Key();
-    std::shared_ptr<const vector<byte>> privKey = std::make_shared<const std::vector<byte>>(
-        (byte*)_keyPair.secret().data(), (byte*)_keyPair.secret().data() + 32);
-    key.setPrivateKey(privKey);
+    if (_keyPair.isInternalKey() == true)
```
[the remark this answers is not in the document]
Comment (MaggieNgWu) [reply libdevcrypto/hsm/HSMSignature.cpp:47]: yes

[the remark this answers is not in the document]
Comment (MaggieNgWu) [reply libdevcrypto/hsm/HSMSignature.cpp:47]: you are right

File: libdevcrypto/hsm/HSMSignature.cpp (hunk 4 above)
Comment (MaggieNgWu) [reply libdevcrypto/hsm/HSMSignature.cpp:103]: Sure

**Pull request merged**